perception/image_projection_based_fusion/src/fusion_node.cpp

```diff
@@ -157,67 +143,157 @@ void FusionNode<Msg, Obj>::preprocess(Msg & ouput_msg __attribute__((unused)))
-      default:
-        RCLCPP_ERROR(this->get_logger(), "invalid image id. id is %zu", image_id);
-        return;
-    }
+  std::lock_guard<std::mutex> lock(mutex_);
+  auto period = std::chrono::duration_cast<std::chrono::nanoseconds>(
+    std::chrono::duration<double>(timeout_ms_));
+  try {
+    setPeriod(period.count());
+  } catch (rclcpp::exceptions::RCLError & ex) {
+    RCLCPP_WARN_THROTTLE(get_logger(), *get_clock(), 5000, "%s", ex.what());
+  }
+  timer_->reset();
+
+  stop_watch_ptr_->toc("processing_time", true);
+
+  typename Msg::SharedPtr output_msg = std::make_shared<Msg>(*input_msg);
 
-    if (camera_info_map_.find(image_id) == camera_info_map_.end()) {
-      RCLCPP_WARN(this->get_logger(), "no camera info. id is %zu", image_id);
+  preprocess(*output_msg);
+
+  int64_t timestamp_nsec =
+    (*output_msg).header.stamp.sec * (int64_t)1e9 + (*output_msg).header.stamp.nanosec;
+
+  // if matching rois exist, fuseonsingle
+  for (std::size_t roi_i = 0; roi_i < rois_number_; ++roi_i) {
+    if (camera_info_map_.find(roi_i) == camera_info_map_.end()) {
+      RCLCPP_WARN(this->get_logger(), "no camera info. id is %zu", roi_i);
       continue;
     }
-    if (debugger_) {
-      debugger_->clear();
+    if ((roi_stdmap_.at(roi_i)).size() > 0) {
+      int64_t min_interval = 1e9;
+      int64_t matched_stamp = -1;
+      std::list<int64_t> outdate_stamps;
+
+      for (const auto & [k, v] : roi_stdmap_.at(roi_i)) {
+        int64_t newstamp = timestamp_nsec + input_offset_ms_.at(roi_i) * (int64_t)1e6;
+        int64_t interval = abs(int64_t(k) - newstamp);
+
+        if (interval <= min_interval && interval <= match_threshold_ms_ * (int64_t)1e6) {
+          min_interval = interval;
+          matched_stamp = k;
+        } else if (int64_t(k) < newstamp && interval > match_threshold_ms_ * (int64_t)1e6) {
+          outdate_stamps.push_back(int64_t(k));
+        }
+      }
+
+      // remove outdated stamps
+      for (auto stamp : outdate_stamps) {
+        (roi_stdmap_.at(roi_i)).erase(stamp);
+      }
+
+      // fuseonSingle
+      if (matched_stamp != -1) {
+        if (debugger_) {
+          debugger_->clear();
+        }
+
+        fuseOnSingleImage(
+          *input_msg, roi_i, *((roi_stdmap_.at(roi_i))[matched_stamp]), camera_info_map_.at(roi_i),
+          *output_msg);
+        (roi_stdmap_.at(roi_i)).erase(matched_stamp);
+        is_fused_.at(roi_i) = true;
+
+        // add timestamp interval for debug
+        if (debug_publisher_) {
+          double timestamp_intervel_ms = (matched_stamp - timestamp_nsec) / 1e6;
+          debug_publisher_->publish<tier4_debug_msgs::msg::Float64Stamped>(
+            "debug/timestamp_intervel_ms/roi" + std::to_string(roi_i), timestamp_intervel_ms);
+          debug_publisher_->publish<tier4_debug_msgs::msg::Float64Stamped>(
+            "debug/timestamp_intervel_offset_ms/roi" + std::to_string(roi_i),
+            timestamp_intervel_ms - input_offset_ms_.at(roi_i));
```

Comment:
same as <https://github.com/autowarefoundation/autoware.universe/pull/2175/files#r1015138919>

Reply:
fixed in https://github.com/autowarefoundation/autoware.universe/pull/2175/commits/7807a448f3c772d49ebe50bc6f5fe3863ead6a64